### ISSUE TITLE
How shall we combine effort with red guide?

### ISSUE THREAD

**Issue by josephholsten** [sous-chefs/meta#1]
http://github.com/redguide is an already existing group like ours. Should we just move everything into redguide?

**Comment by luishdez** [comment]
It doesn't look that they have a big team and "structure" already defined. In fact there are more people in this group already. May be they should move here…

I think I would define a manifesto and code/naming/style guide for basic things before starting adopt other projects though

**Comment by guilhem** [comment]
@luishdez in fact we are 11 and chef-brigade 15 :) that's not a big difference.
And this is 11 without any announcement on chef mailing-list ;)

Redguide is already quite old and we already manage some problems in past.
For example, we have done a lot of work about nodejs, they was 3 different cookbooks and maintainers and we manage to unified them around 1.
We are currently doing this work on "locales" cookbook.

This is a technical work, but also a human problem: contacting maintainer, get rights on supermarket etc.

We also have defined guideline for maintaining cookbooks, it's not published yet (may be done today ;)) but any new member is briefed on it.
We may discuss about it, but it seems to work quite well :)

**Comment by guilhem** [comment]
And yes, we lack people in "redguide owner" org who have same view as us: quality and progress :)

(today we have 1 team for each cookbook and 1 team owner to manage any problem or adding new cookbooks)

We will be really happy to see you guys helping us :)

**Comment by luishdez** [comment]
@guilhem sorry if I made a poor judgment, Happy to colab with you guys colab is always good

**Comment by bkconrad** [comment]
I'm all for combining efforts. My only opinion on the matter is that "chef-brigade" is a pretty cool name :)

**Comment by zarry** [comment]
agreed with @bkconrad, I am not opposed to joining efforts but I am a bit partial to chef-brigade :+1:

**Comment by fonsecas72** [comment]
My 2cents
Redguide is already a mature organization, it already has a plan, a structure, a website, while Chef Brigade really sounds better and has more users.
I can't really see one clearly better than the other and I don't think it makes sense for both to coexist.
But If none is willing to "leave" his cause what about joining forces in a new organization? Maybe "Friends of Chef" ?

**Comment by fonsecas72** [comment]
We would use the best of both orgs off course.

**Comment by josephholsten** [comment]
I'm going to ignore the issues of naming and membership. I'll gladly accept the ridiculous titles of brigadier and bibendum :wink:

Can we all agree to sharing the same policy documents? I think we're mostly concerned with conditions of adopting a new cookbook, and what to do when a maintainer has to step away. 

Also, can we agree to not adopt conflicting cookbooks? Eg if redguide begins to support a redis cookbook, chef-bridage will not adopt a different redis cookbook, and anything that depends on a redis cookbook will try to use this common cookbook.

If we agree to that, I'd put together a PR with some words to that effect for this repo's readme.

**Comment by BarthV** [comment]
Hi ! (I'm from @redguide :wink: )

I'm absolutely ok with sharing our rules and best practice :-). with @guilhem we're targeting a "foodcritic" or RFC rules approach which can be easily shared and modified/upgraded. But as a "young" org, you can make whatever you want. We'll adapt :)

My opinion is the rules are really not the key here ... It's sooo much more about human relationship.
For example, I just met tonight a typical problem that we all seek to avoid :
https://github.com/dcm-ops/chef-slack_handler
- it's a very young but helpful cookbook (8 commit)
- this cookbook is not moving since november 2014.
- there's few PR and issues.
- rackspace team have a PR since december and they decided to put their own PR branch on Supermarket. (they merged it later on)
  - https://github.com/rackspace-cookbooks/chef-slack_handler
  - https://supermarket.chef.io/cookbooks/slack_handler
- upstream project moved forward and now they ask to publish on Supermarket
  - https://github.com/dcm-ops/chef-slack_handler/issues/5

What now ? 
- I have to try to mediate between all stakeholders and seek to both authors the right to "take control", explain the benefits to participate in "the game" but also ask Rackspace to give back the ownership on the Supermarket namespace (the hardest part IMO)...
  - https://github.com/dcm-ops/chef-slack_handler/pull/6#issuecomment-102706599

It is absolutely not a technical job, neither a rewarding job, or even a nice work to do. There's no "rules" about that part ... I'm probably wasting my time and I could totally rewrite the entire cookbook in one night (and make a better one)...
But each of you and us will always have to remeber that we have to do the same exact thing to make our community the best (and avoid namespace conflict ... just look at Ansible Galaxy mess. I don't want that for Chef) !

So just fight for your brigade, you'll always need bibendum for your tires after several kilometers.

**Comment by martinb3** [comment]
Perhaps it's worth considering your target audience as well. Are you seeking to take over cookbooks that are already maintained, or simply maintain orphaned/abandoned cookbooks?

We've just had @BarthV visit one of our forks of a repository for a cookbook that was completely unmaintained, that we forked/improved (our PR has never been merged), and started doing releases of, including releases to Supermarket. He's accused us of somehow 'taking' the Supermarket name (it's been vacant for over a year), just because we're willing to review and merge community PRs and release new versions.

> upstream project moved forward and now they ask to publish on Supermarket: dcm-ops/chef-slack_handler#5

Not true at all. That request was from the same time period as our original PR. The project has _not_ moved forward. The last commit to that repo was Nov 1, 2014. The issue you reference is also from 2014.

I think a focus on unmaintained cookbooks is a wonderful thing, but we've volunteered to maintain this cookbook, and now we're having someone tell us @redguide or @chef-brigade should do it and we've somehow stolen the name in the namespace. I think trying to accuse volunteers of squatting on names is probably not in the best benefit of the community.

I personally (not speaking as my employer) would gladly participate in either effort, @reguide or @chef-brigade, but shouldn't we start with cookbooks that are truly abandoned? And shouldn't we approach authors or maintainers of the Supermarket names with care, as opposed to saying "you stole this, give it back"?

It feels like @BarthV's comments were just trolling for control of the Supermarket namespace, as opposed to actually improving the maintenance of cookbooks in the community.

**Comment by dblessing** [comment]
@martinb3 If upstream is dead and there's an active fork then I don't believe a group like this has any reason to come and snap it up. Any movement between upstream and fork needs to happen between the upstream maintainer and the fork.

Let's try not to put so much bureaucracy in this effort. Have a mission, some general guiding terms, maybe a code of conduct and let the rest happen organically.

**Comment by dblessing** [comment]
Whoops. Didn't mean to close :)

**Comment by guilhem** [comment]
Main problem in supermarket is governance, especially when cookbook is maintain by a company / single user.

How to deal with new PR, breaking changes... That's why we created `redguide`.

Our main focus is quality to being able to be used by company for their daily prod.
But also being able to do changes who needs to be done without any pressure from manager or CTO.

And... it works!

I really have bad memories of the transfer from Opscode to individual for many cookbooks.
I don't want it to happen again.

In fact, redguide have no goal to support all crappy orphan cookbooks that can be found on supermarket (and there are many from me ;)), but goal is to support heavy used cookbook with open governance.

If chef-guard goal is to support any cookbook, so both org can live their own life :)

**Comment by josephholsten** [comment]
seems like we're in agreement, redguide and chef-brigade will continue to be seperate, but will try to work together whenever possible.
